### PR TITLE
Display "You" instead of own phone number

### LINKED ISSUE #4411: Creating a new group lists your phone number
When creating a new group (in this case of 3 users including yourself) the text is displayed as:

>[Your phone number], Username1, Username2 joined the group. Title is now 'name'.

I'm proposing this is changed to:
>You, Username1, Username2 joined the group....

This makes the text line shorter, clearer, and less confusing for those that don't actually memorize their own phone numbers.

### PR DESCRIPTION
**Pull request by cascheberg**
### Contributor checklist

- [x] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [x] I have tested my contribution
- [x] My contribution is fully baked and ready to be merged as is
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message

---
### Description

Fixes #4411

**Before:**
GroupYouBefore

**After:**
GroupYouAfter

// FREEBIE

### CODE REVIEW

**Comment by RiseT** [comment]
"You" might be prefered to "Me":

https://material.google.com/style/writing.html#writing-language

**Comment by cascheberg** [comment]
Updated "Me" -> "You"

**Comment by RiseT** [comment]
You might want to change the PR title accordingly.

**Comment by kevinwestern** [comment]
How often are PRs merged? It looks like you've sent quite a few PRs but most are still pending. I'd like to contribute, but I don't want the work to be wasted if it's never merged.

File: src/org/thoughtcrime/securesms/MediaPreviewActivity.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ private void initializeActionBar() {
     } else {
       relativeTimeSpan = null;
     }
-    getSupportActionBar().setTitle(recipient == null ? getString(R.string.MediaPreviewActivity_you)
+    getSupportActionBar().setTitle(recipient == null ? getString(R.string.Recipients_you)
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/MediaPreviewActivity.java]: what's the rationale for this change?

File: src/org/thoughtcrime/securesms/PassphraseRequiredActionBarActivity.java [hunk 2]
```diff
@@ -152,9 +150,8 @@ private int getApplicationState(MasterSecret masterSecret) {
       return STATE_UPGRADE_DATABASE;
     } else if (!TextSecurePreferences.hasPromptedPushRegistration(this)) {
       return STATE_PROMPT_PUSH_REGISTRATION;
-    } else {
-      return STATE_NORMAL;
     }
+    return STATE_NORMAL;
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/PassphraseRequiredActionBarActivity.java]: why is this necessary?

File: src/org/thoughtcrime/securesms/recipients/Recipients.java [hunk 3]
```diff
@@ -285,17 +288,18 @@ public String getSortedIdsString() {
...
+    LinkedList<String> from = new LinkedList<>();
 
-    for (int i=0;i<recipients.size();i++) {
-      fromString += recipients.get(i).toShortString();
-
-      if (i != recipients.size() -1 )
-        fromString += ", ";
+    for (Recipient recipient : recipients) {
+      if (Util.isOwnNumber(context, recipient.getNumber())) {
+        from.push(context.getString(R.string.Recipients_you));
```
Comment (moxie0) [inline src/org/thoughtcrime/securesms/recipients/Recipients.java]: This feels too low level for me for this change

File: src/org/thoughtcrime/securesms/MediaPreviewActivity.java (hunk 1 above)
Comment (cascheberg) [inline src/org/thoughtcrime/securesms/MediaPreviewActivity.java]: It's the same string "You", so I figured it could use the same id. In the German Signal translation for instance these are translated differently, which could have been avoided.

File: src/org/thoughtcrime/securesms/PassphraseRequiredActionBarActivity.java (hunk 2 above)
Comment (cascheberg) [inline src/org/thoughtcrime/securesms/PassphraseRequiredActionBarActivity.java]: This was left over, I can revert it.

File: src/org/thoughtcrime/securesms/recipients/Recipients.java (hunk 3 above)
Comment (cascheberg) [inline src/org/thoughtcrime/securesms/recipients/Recipients.java]: Do you have another suggestion?
I think I remember that handling it inside `recipient` was much more difficult. This solution has the nice effect that "you" is always at the beginning of the string. Or is it something else that you would like to change?

**Comment by cascheberg** [comment]
Another straight forward solution would be to remove your own number from the list of contacts that where added to the group and change the wording to "User1, User2 also joined the group."

Seeing your own entryg as just a phone number is really not great.

**Comment by altring** [comment]
When will we see this?

**Comment by alan-signal** [comment]
Thank you!

Unfortunately we will not merge this as this issue has since been resolved. Groups now list "Me".

**Comment by cascheberg** [comment]
@alan-signal the issue still exists. It is about the group update message that is shown when a group is created. I created #8888 to document it.
This PR is quite old and I did not care to maintain it because apparently no one was interest in merging it, but it's nice to see that community PRs are getting some attention again.

**Comment by alan-signal** [comment]
I miss-understood, I'll reopen and give it a closer look.

**Comment by cascheberg** [comment]
@alan-signal Thanks. As I understood from Moxie this change was too low-level for a rather small improvement. Let me know what you think. I can rebase or rework it.

**Comment by alan-signal** [comment]
Thanks but I think it might have been fixed, will close again when it makes it to public.

**Comment by Mark5595** [comment]
Hi there, 

Seems like this issue still exists for group MMS messages. It appears in 3 views that my own number is displayed as a raw phone number rather than a more helpful `me`, `you` or even nothing. 

1. Conversation list (main screen on app reset) view where you can jump into a conversation shows my number in a group MMS conversation.

2. Clicking into a group MMS conversation shows my number at the header / title bar 

3. From a group MMS clicking the header (next to the back button) opening up the the chat settings view shows my number 

The one place where my raw number is not exposed is when clicking on the people icon from a group MMS screen there the `Group Members` modal opens and appropriately shows `me` instead of my raw number.   

Displaying my number takes up valuable screen real estate where I would rather see my peers in the conversation as I know I am in a conversation that I can see (otherwise it wouldn't show up).

**Comment by cascheberg** [comment]
@alan-signal I noticed that there have been changes regarding PROFILE_DISPLAY, but the issue described and fixed here still remains in the current release. I made a rebase on the current master, please check again if this is still useful. I added screenshots to the original description. The commit is much simpler now than it was three years ago, because some required changes have already been implemented lately.
